### PR TITLE
Destructors: more moves for tuples

### PR DESCRIPTION
**Pull request by cooldome**
I have found a couple more use cases where sink is not happening for tuples.
nkStmtListExpr in moveOrCopy is likely going to help not only tuples.

### CODE REVIEW

File: compiler/destroyer.nim [hunk 1]
```diff
@@ -458,6 +458,11 @@ proc moveOrCopy(dest, ri: PNode; c: var Con): PNode =
     else:
       result = genCopy(c, dest.typ, dest, ri)
     result.add p(ri, c)
+  of nkStmtListExpr:
+    result = newNodeI(nkStmtList, ri.info)
```
Comment (Araq) [inline compiler/destroyer.nim]: But that's dubious, it should be at least ``newNodeIT(nkStmtListExpr, ri.info, ri.typ)``

Comment (cooldome) [reply compiler/destroyer.nim]: IMO, I am  rewriting
```nim
a = (
   expr1
   expr2
   expr3)
```
into
```
expr1
expr2
a = expr3
```
Pretty much expression to statement lowering. In similar fashion I thought of lowering ifExpr to ifStmt, case expr to case stmt, block expr to block statement by replacing their last operations in branches.Mentioned operations are still always copying.

Do you have alternative idea?

**Comment by cooldome** [comment]
Hi Araq, let me know what you think. I will have some time later today to sort it out and would like to fix also if, case expressions as well

File: compiler/destroyer.nim (hunk 1 above)
Comment (Araq) [reply compiler/destroyer.nim]: Ok, I checked that ``moveOrCopy`` is not supposed to produce an expression context, so your patch is fine.

**Pull request merged**